src/main/java/com/google/devtools/build/lib/rules/proto/ProtoCommon.java

```diff
@@ -568,37 +525,27 @@ public static ProtoInfo createProtoInfo(
     return outputsBuilder.build();
   }
 
-  /**
-   * Returns the .proto files that are the direct srcs of the direct-dependencies of this rule. If
-   * the current rule is an alias proto_library (=no srcs), we use the direct srcs of the
-   * direct-dependencies of our direct-dependencies.
-   */
-  @Nullable
-  private static NestedSet<Pair<Artifact, String>> computeStrictImportableProtos(
-      ImmutableList<ProtoInfo> protoDeps,
-      ImmutableList<Pair<Artifact, String>> sourceImportPathPairs) {
-    NestedSetBuilder<Pair<Artifact, String>> result = NestedSetBuilder.stableOrder();
-    if (sourceImportPathPairs.isEmpty()) {
-      for (ProtoInfo provider : protoDeps) {
-        result.addTransitive(provider.getStrictImportableProtoSourcesImportPaths());
-      }
-    } else {
-      for (ProtoInfo provider : protoDeps) {
-        result.addTransitive(provider.getStrictImportableProtoSourcesImportPathsForDependents());
+  private static NestedSet<ProtoSource> computeStrictImportableProtos(
+      ImmutableList<ProtoSource> directSources, ImmutableList<ProtoInfo> deps) {
+    NestedSetBuilder<ProtoSource> builder = NestedSetBuilder.stableOrder();
+    if (!directSources.isEmpty()) {
+      builder.addAll(directSources);
+      for (ProtoInfo provider : deps) {
+        builder.addTransitive(provider.getExportedSources());
       }
-      result.addAll(sourceImportPathPairs);
     }
-    return result.build();
+    return builder.build();
   }
 
   /**
    * Returns the .proto files that are the direct srcs of the exported dependencies of this rule.
    */
-  private static NestedSet<Pair<Artifact, String>> computeExportedProtos(
-      ImmutableList<ProtoInfo> exports) {
-    NestedSetBuilder<Pair<Artifact, String>> result = NestedSetBuilder.stableOrder();
-    for (ProtoInfo provider : exports) {
-      result.addTransitive(provider.getStrictImportableProtoSourcesImportPaths());
+  private static NestedSet<ProtoSource> computePublicImportProtos(
+      ImmutableList<ProtoSource> directSources, ImmutableList<ProtoInfo> exports) {
+    NestedSetBuilder<ProtoSource> result = NestedSetBuilder.stableOrder();
+    result.addAll(directSources);
```

Comment:
This line was not present in `computeExportedProtos()` and is probably the cause of the aforementioned test breakage